pkg/collector/corechecks/oracle-dbm/activity.go

```diff
@@ -101,6 +101,87 @@ WHERE
 	)
 	AND status = 'ACTIVE'`
 
+const ACTIVITY_QUERY_RDS = `SELECT /* DD_ACTIVITY_SAMPLING */
+s.sid,
+s.serial#,
+s.username,
+s.status,
+s.osuser,
+s.process,
+s.machine,
+s.port,
+s.program,
+s.type,
+s.sql_id,
+sq.force_matching_signature as force_matching_signature,
+sq.plan_hash_value sql_plan_hash_value,
+s.sql_exec_start,
+s.sql_address,
+s.prev_sql_id,
+sq_prev.plan_hash_value prev_sql_plan_hash_value,
+s.prev_exec_start as prev_sql_exec_start,
+sq_prev.force_matching_signature as prev_force_matching_signature,
+s.prev_sql_addr prev_sql_address,
+s.module,
+s.action,
+s.client_info,
+s.logon_time,
+s.client_identifier,
+CASE WHEN blocking_session_status = 'VALID' THEN
+blocking_instance
+ELSE
+null
+END blocking_instance,
+CASE WHEN blocking_session_status = 'VALID' THEN
+  blocking_session
+ELSE
+  null
+END blocking_session,
+CASE WHEN final_blocking_session_status = 'VALID' THEN
+  final_blocking_instance
+ELSE
+  null
+END final_blocking_instance,
+CASE WHEN final_blocking_session_status = 'VALID' THEN
+  final_blocking_session
+ELSE
+  null
+END final_blocking_session,
+CASE WHEN state = 'WAITING' THEN
+  event
+ELSE
+  'CPU'
+END event,
+CASE WHEN state = 'WAITING' THEN
+  wait_class
+ELSE
+  'CPU'
+END wait_class,
+s.wait_time_micro,
+c.name as pdb_name,
+sq.sql_fulltext as sql_fulltext,
+sq_prev.sql_fulltext as prev_sql_fulltext,
+comm.command_name
+FROM
+v$session s,
+v$sql sq,
+v$sql sq_prev,
+v$containers c,
+v$sqlcommand comm
+WHERE
+sq.sql_id(+)   = s.sql_id
+AND sq.child_number(+) = s.sql_child_number
+AND sq_prev.sql_id(+)   = s.prev_sql_id
+AND sq_prev.child_number(+) = s.prev_child_number
+AND ( sq.sql_text NOT LIKE '%DD_ACTIVITY_SAMPLING%' OR sq.sql_text is NULL ) 
```

Comment:
`sq.sql_text NOT LIKE '%DD_ACTIVITY_SAMPLING%'` would it be better / more efficient to filter on the SID than doing string comparison to filter out this session?

Reply:
Not sql_id but session ID. Something along the lines of this to filter out the current session issuing the query:

```
AND sid != (select sid from v$mystat where rownum=1);
```
?

Reply:
But the session filter would exclude all agent queries. We want to exclude only the activity sampling query.